### PR TITLE
Added hint text improvements based on designs

### PR DESCRIPTION
**Pull request by mkerns1**
## Summary

- Update hint text and field label for the First Name fields of both Veteran and Caregivers in the form.
- Update hint text and field label for the Street Address fields of both Veteran and Caregivers.
- Modify the address callout label on both the Primary Caregiver Info and Secondary Caregiver Info pages to align with the updated Street Address field label.

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#60229

## Acceptance criteria
- Content matches UX declared text

### CODE REVIEW

**Review by JoshingYou1 — APPROVED** [review]
LGTM! Nice work Matt!!! 😃

File: src/applications/caregivers/definitions/UIDefinitions/sharedUI.js [hunk 1]
```diff
@@ -35,9 +39,10 @@ export const genderUI = label => ({
 
 export const fullNameUI = label => ({
   first: {
-    'ui:title': `${label} first name`,
+    'ui:title': `${label} legal first name`,
+    'ui:description': VeteranFullNameDescription,
```
Comment (longmd) [inline src/applications/caregivers/definitions/UIDefinitions/sharedUI.js:43]: You're going to want to take this out of here and extend the definition in the ui:schema declaration. This is used whenever there is a name field (like for primary and secondary caregiver) so the veteran description will not apply in those cases. See how the SSN is extended in `vetInfo.js` form config file.

File: src/applications/caregivers/definitions/UIDefinitions/sharedUI.js [hunk 2]
```diff
@@ -76,11 +81,12 @@ export const addressWithoutCountryUI = label => ({
   'ui:title': ' ',
   'ui:order': ['street', 'street2', 'city', 'state', 'postalCode'],
   street: {
-    'ui:title': `${label} current street address`,
-    'ui:errorMessages': { required: 'Please enter a street address' },
+    'ui:title': `${label} current home address`,
+    'ui:description': VeteranHomeAddressDescription,
```
Comment (longmd) [inline src/applications/caregivers/definitions/UIDefinitions/sharedUI.js:85]: Same thing here, as mentioned above. You will need to extend the declaration where it is used and not bake this into the shared schema declaration.

**Review by longmd — CHANGES_REQUESTED** [review]
Aside from the two comments related to baking the description into that default declaration, I wanted to check to see if the addition of "legal" needs to apply to all people's names? Are we adding that to the name fields for the caregivers, too, or just the Veteran?

File: src/applications/caregivers/definitions/UIDefinitions/sharedUI.js (hunk 1 above)
Comment (mkerns1) [reply src/applications/caregivers/definitions/UIDefinitions/sharedUI.js]: Hey Matt let me know how this looks. I made it more dynamic based on Jerek's suggestion. Now the new customFullNameUI can extend a specific field within a UI schema object with additional properties.

**Comment by mkerns1** [comment]
> Aside from the two comments related to baking the description into that default declaration, I wanted to check to see if the addition of "legal" needs to apply to all people's names? Are we adding that to the name fields for the caregivers, too, or just the Veteran?

From Design: "Yes, we are changing the field label for all people submitting the CG to keep the fields consistent throughout the form"

**Review by longmd — CHANGES_REQUESTED** [review]
Can we add some unit testing for the new shared function? I know there probably isn't a ton of stuff existing that tests the other functions in that file, but we need to get into a habit of adding testing for new things that we do.

**Comment by mkerns1** [comment]
> Can we add some unit testing for the new shared function? I know there probably isn't a ton of stuff existing that tests the other functions in that file, but we need to get into a habit of adding testing for new things that we do.

Definitely not a problem. Let me know how that looks and if I need to add anything to it!

**Review by longmd — APPROVED** [review]

**Pull request merged**